src/DotLiquid/Strainer.cs

```diff
@@ -15,6 +15,27 @@ static class DictionaryExtensions
                 return dic[key] = factory();
             return found;
         }
+
+    }
+
+    static class MethodInfoExtensions
+    {
+        public static int GetNonContextParameterCount(this MethodInfo method)
+        {
+            return method.GetParameters().Count(p => p.ParameterType != typeof(Context));

```

Comment:
avoid single letter variable names for clarity. Always use complete words from English dictionary, unless there is a well-accepted acronym.

Reply:
And yes we do have existing code that violates this, not particularly easy to decypher.

Reply:
Fixed: https://github.com/dotliquid/dotliquid/pull/453/commits/72a2410fe218c54f5da077e7012d54d35c87b34c